cartography/intel/azure/function_app.py

```diff
@@ -0,0 +1,707 @@
+
+
+@timeit
+def get_function_apps_list(credentials: Credentials,
+                           subscription_id: str) -> List[Dict]:
+    try:
+        client = get_client(credentials, subscription_id)
+        function_app_list = list(
+            map(lambda x: x.as_dict(), client.web_apps.list()))
+        for function in function_app_list:
+            x = function['id'].split('/')
+            function['resource_group'] = x[x.index('resourceGroups') + 1]
+
+        return function_app_list
+
+    except HttpResponseError as e:
+        logger.warning(f"Error while retrieving function apps - {e}")
+        return []
+
+
+def _load_function_apps_tx(tx: neo4j.Transaction, subscription_id: str,
+                           function_apps_list: List[Dict],
+                           update_tag: int) -> None:
+    ingest_function_apps = """
+    UNWIND {function_apps_list} AS function_app
+    MERGE (f:AzureFunctionApp{id: function_app.id})
+    ON CREATE SET f.firstseen = timestamp(),
+    f.type = function_app.type,
+    f.location = function_app.location,
+    f.resourcegroup = function_app.resource_group
+    SET f.lastupdated = {update_tag},
+    f.name = function_app.name,
+    f.container_size = function_app.container_size,
+    f.default_host_name=function_app.default_host_name,
+    f.last_modified_time_utc=function_app.last_modified_time_utc,
+    f.state=function_app.state,
+    f.repository_site_name=function_app.repository_site_name,
+    f.daily_memory_time_quota=function_app.daily_memory_time_quota,
+    f.availability_state=function_app.availability_state,
+    f.usage_state=function_app.usage_state
+    WITH f
+    MATCH (owner:AzureSubscription{id: {SUBSCRIPTION_ID}})
+    MERGE (owner)-[r:RESOURCE]->(f)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {update_tag}
+    """
+
+    tx.run(
+        ingest_function_apps,
+        function_apps_list=function_apps_list,
+        SUBSCRIPTION_ID=subscription_id,
+        update_tag=update_tag,
+    )
+
+
+def cleanup_function_apps(neo4j_session: neo4j.Session,
+                          common_job_parameters: Dict) -> None:
+    run_cleanup_job('azure_import_function_apps_cleanup.json', neo4j_session,
+                    common_job_parameters)
+
+
+def sync_function_apps(
+    neo4j_session: neo4j.Session,
+    credentials: Credentials,
+    subscription_id: str,
+    update_tag: int,
+    common_job_parameters: Dict,
+) -> None:
+    function_apps_list = get_function_apps_list(credentials, subscription_id)
+    load_function_apps(neo4j_session, subscription_id, function_apps_list,
+                       update_tag)
+    cleanup_function_apps(neo4j_session, common_job_parameters)
+
+
+def get_function_apps_configuration_list(credentials: Credentials,
+                                         subscription_id: str) -> List[Dict]:
+    try:
+        client = get_client(credentials, subscription_id)
+        function_apps_list = get_function_apps_list(credentials,
```

Comment:
instead of getting the function apps again here, we should pass the function apps as a parameter to the `get_function_apps_configuration_list` function. function apps are already retrieved as part of `sync_function_apps`.